### ISSUE TITLE
Offset error when semicolons are disallowed

### ISSUE THREAD

**Issue by nichtich** [peggyjs/peggyjs-eslint-plugin#37]
Again, minimal example at https://github.com/nichtich/test-peggy-eslint

Adding a rule to disallow semicolon will result in an error:

~~~js
module.exports = {
  extends: ["plugin:@peggyjs/recommended"],
  rules: {
    semi: [ "error", "never" ], // Disallow semicolons
  },
};
~~~

with this grammar:

~~~
X = "x" { return "x"; }
~~~

It's `Error: Invalid offset: 510`

When the semicolon is removed, the offset is 509.

**Comment by hildjj** [comment]
OK, there are two issues here.  The first is that we're trying to do a fix on a portion of the generated JS that is after the last user-created portion of that synthetic file.  That's a quick fix.

The other is that we're allowing through errors from outside user-created javascript.  So, once I fix the initial error above, there are actually 16 semi errors reported (all but one from the template code), not just one.

**Comment by nichtich** [comment]
I think it is best practice to exclude the generated JS from linting because it is not meant to be touched by humans anyway, but separation of generated JS and user-created JS seems tricky. I [tried to limit the rule to .js files only](https://github.com/nichtich/test-peggy-eslint/commit/9f6b1b8f6a393554ea7a8a55968ee13868712701) without success.

**Comment by hildjj** [comment]
I agree, and I've found a way to do it.  Patch almost ready.

**Comment by hildjj** [comment]
The temporary file name for `foo.peggy` is `foo.peggy/0.js`.

**Comment by hildjj** [comment]
Fixed in v2.1.2
